packages/flutter/test/material/raised_button_test.dart

```diff
@@ -617,6 +617,57 @@ void main() {
     expect(box.size, equals(const Size(76, 36)));
     expect(childRect, equals(const Rect.fromLTRB(372.0, 293.0, 428.0, 307.0)));
   });
+
+  testWidgets('RaisedButton.icon responds to applied padding', (WidgetTester tester) async {
+    const Key buttonKey = Key('test');
+    const Key labelKey = Key('label');
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: RaisedButton.icon(
+              icon: const Icon(Icons.add),
+              padding: const EdgeInsets.all(16),
+              key: buttonKey,
+              onPressed: () {},
+              label: const Text(
+                'Hello',
+                key: labelKey,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Size buttonSize = tester.getSize(find.byKey(buttonKey));
+    final Size labelSize = tester.getSize(find.byKey(labelKey));
+    final Rect paddingRect = tester.getRect(find.byType(Padding));
+    final Rect labelRect = tester.getRect(find.byKey(labelKey));
+    final Rect iconRect = tester.getRect(find.byType(Icon));
+
+    // Check the RaisedButton.icon size
+    // By default the padding between icon and label is 8.0, and default
+    // size of Icon is 24 pixels * 24 pixels, hence
+    // height = 24 + 16 * 2 = 56 and
+    // width = 24 + width of Text + 8 + 16 * 2
+    expect(buttonSize, Size(labelSize.width + 8 + 24 + 32, 56));
+
+    // Check the position of Text and Icon
+    expect(paddingRect.right, labelRect.right + 16);
+    // The height of Icon is 24 pixels which is more than Text
+    // Hence, we will subtract Icon Height - Text Height and divide by 2
+    expect(paddingRect.top, labelRect.top - (16 + (24-labelSize.height)/2));
+    expect(paddingRect.bottom, labelRect.bottom + (16 + (24-labelSize.height)/2));
+    // By default the padding between icon and label is 8.0,
+    // Icon width is 24 pixels and padding is 16, hence 8.0 + 16.0 + 24.0 = 48
+    expect(paddingRect.left,labelRect.left - 48);
+    expect(paddingRect.left,iconRect.left-16);
+    expect(paddingRect.top,iconRect.top-16);
+    expect(paddingRect.bottom,iconRect.bottom+16);
+    expect(paddingRect.right, iconRect.right + 8 + 16 + labelSize.width);
```

Comment:
I don't think all of these checks are necessary. Something like this will suffice:
```dart
    // Icon is the left-most widget in a [RaisedButton] when textDirection is set to [TextDirection.ltr].
    // Compare padding with [Icon] height, since the [Icon] is taller than the label [Text] widget.
    expect(paddingRect.left, iconRect.left - 16);
    expect(paddingRect.top, iconRect.top - 16);
    expect(paddingRect.right, labelRect.right + 16);
    expect(paddingRect.bottom, iconRect.bottom + 16);
```

However, you could probably programmatically determine whether icon and the label is taller and use the taller of the two for comparison. You could write similar logic to determine which widget is the left-most or right-most widget as well.

Reply:
I guess having the logic for which widget is left-most or right-most is not need since we have already define text direction. WDYT? 
I'll surely add the logic for checking which widget is taller and do the test accordingly.